tests/data/test_converter_public_trades.py

```diff
@@ -0,0 +1,344 @@
+                                       trades=trades_red,
+                                       plot_config=plotconf
+                                       )
+    pyo.plot(graph, output_type="file", show_link=False,
+             filename="tests/data/test_converter_public_trades.html")
+
+
+# need to be at last to see if some test changed the testdata
+# always need to use .copy() to avoid changing the testdata
+def test_public_trades_testdata_sanity(candles, trades, public_trades_list, public_trades_list_simple,
+                                       public_trades_list_simple_bidask, public_trades_list_simple_results,
+                                       populate_dataframe_with_trades_dataframe, populate_dataframe_with_trades_trades):
+    assert 10999 == len(candles)
+    assert 1811 == len(trades)
+    assert 1000 == len(public_trades_list)
+    assert 3 == len(public_trades_list_simple_results)
+    assert 7 == len(public_trades_list_simple_bidask)
+    assert 999 == len(populate_dataframe_with_trades_dataframe)
+    assert 8033249 == len(populate_dataframe_with_trades_trades)
+
+    assert 7 == len(public_trades_list_simple)
+    assert 5 == public_trades_list_simple.loc[
+        public_trades_list_simple['side'].str.contains(
+            'sell'), 'id'].count()
+    assert 2 == public_trades_list_simple.loc[
+        public_trades_list_simple['side'].str.contains(
+            'buy'), 'id'].count()
+
+    assert public_trades_list.columns.tolist() == [
+        'timestamp', 'id', 'type', 'side', 'price',
+        'amount', 'cost', 'date']
+
+    assert public_trades_list.columns.tolist() == [
+        'timestamp', 'id', 'type', 'side', 'price', 'amount', 'cost', 'date']
+    assert public_trades_list_simple_results.columns.tolist() == [
+        'level', 'bid', 'ask', 'delta']
+    assert public_trades_list_simple.columns.tolist() == [
+        'timestamp', 'id', 'type', 'side', 'price',
+        'amount', 'cost', 'date']
+    assert populate_dataframe_with_trades_dataframe.columns.tolist() == [
+        'date', 'open', 'high', 'low', 'close', 'volume']
+    assert populate_dataframe_with_trades_trades.columns.tolist() == [
+        'timestamp', 'id', 'type', 'side', 'price',
+        'amount', 'cost', 'date']
+
+    public_trades_list_simple_results = pd.DataFrame([[0, 0, 0, 0], [23437.5, 0.245, 0.0, -0.245], [23438.0, 0.0, 0.14, 0.140]],
+                                                     columns=public_trades_list_simple_results.columns)
+    pd.testing.assert_series_equal(
+        public_trades_list_simple_results['delta'], public_trades_list_simple_results['delta'], check_index=False)
+    assert public_trades_list_simple_results.values.tolist(
+    ) == public_trades_list_simple_results.values.tolist()
+
+
+class ReporterPlugin:
+    def pytest_sessionfinish(self):
+        print("*** test run reporting finishing")
+
+
+# # invoke self to be able to debug
+if __name__ == "__main__":
```

Comment:
vscode allows you to debug tests just fine  ... ?

Reply:
it's meant to enable running it as single file instead of the whole testsuite. debug probably the wrong word here

Reply:
`pytest tests/<testfile>` ?